tests/v2/validation/charts/gatekeeperallowednamespacespostupgrade_test.go

```diff
@@ -0,0 +1,34 @@
+package charts
+
+import (
+	"strings"
+
+	settings "github.com/rancher/rancher/pkg/settings"
+	namespaces "github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/pkg/environmentflag"
+	"github.com/stretchr/testify/require"
+)
+
+func (n *GateKeeperTestSuite) TestGateKeeperAllowedNamespacesPostUpgrade() {
+
+	subSession := n.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := n.client.WithSession(subSession)
+	require.NoError(n.T(), err)
+
+	if !client.Flags.GetValue(environmentflag.GatekeeperAllowedNamespaces) {
+		n.T().Skip("skipping TestGateKeeperAllowedNamespacesPostUpgrade because GatekeeperAllowedNamespaces flag not set in cattle config")
+	}
+
+	sysNamespaces := settings.SystemNamespaces.Get()
```

Comment:
I don't understand this controller call from the API, can't we use v1 settings by id get here like an end user? [Here's](https://github.com/rancher/rancher/blob/release/v2.7/tests/v2/codecoverage/setuprancher/main.go#L476-L479) an example for it. By doing this we're using the setting controller. If this is a v3 call, `client.Management.Settings.ById` can be the second option.

Reply:
This isn't an API call, we are getting the settings from the package that way it's release specific. Does that make sense? I think Natalie you were advised to do this right?

Reply:
That makes sense to me, I agree with you since this doesn't check anything necessarily about settings but only to use values.

Reply:
left this how it was